packages/taquito/src/contract/interface.ts

```diff
@@ -283,4 +284,34 @@ export interface ContractProvider extends StorageProvider {
    * @param params FailingNoop operation parameter
    */
   failingNoop(params: FailingNoopParams): Promise<FailingNoopOperation>;
+
+  /**
+   *
+   * @description Stake founds. Will sign and inject an operation using the current context
+   *
+   * @returns An operation handle with the result from the rpc node
+   *
+   * @param stake operation parameter
+   */
+  stake(params: StakingParams): Promise<TransactionOperation>;
+
+  /**
+   *
+   * @description Unstake founds. Will sign and inject an operation using the current context
+   *
+   * @returns An operation handle with the result from the rpc node
+   *
+   * @param unstake operation parameter
+   */
+  unstake(params: StakingParams): Promise<TransactionOperation>;
+
+  /**
+   *
+   * @description Finalize unstake founds. Will sign and inject an operation using the current context
```

Comment:
octez-client desc:  `Transfer all the finalizable unstaked funds of the source to their liquid balance.`